src/text-error-reporter.cpp

```diff
@@ -63,7 +63,9 @@ void text_error_formatter::write_message_part(
 void text_error_formatter::write_after_message(std::string_view code,
                                                diagnostic_severity,
                                                const source_code_span &) {
-  this->output_ << " [" << code << "]\n";
+  this->output_ << " ["
+                << "\x1B]8;;https://quick-lint-js.com/errors/" << code
+                << "\x1B\\" << code << "\x1B]8;;\x1B\\]\n";
```

Comment:
Must fix: We shouldn't do this when writing to a pipe or a file. Use `isatty` to check if we are writing to a terminal.

It might be impossible to call `isatty` here since `this->output_` is an `std::ostream`, which might not be backed by a file descriptor. So perhaps we should pass the result of `isatty` to `text_error_reporter`'s constructor.

We might want to provide the option in the constructor anyway if we provide the user the option of enabling or disabling these escape codes. For example, GCC (a C++ compiler) allows the user to choose whether URLs are included for diagnostics using the `-fdiagnostics-urls` option: https://gcc.gnu.org/onlinedocs/gcc-11.2.0/gcc/Diagnostic-Message-Formatting-Options.html

Reply:
Should we do both?
1. User option for enabling or disabling escape codes
2. Check `stdout` with `isatty`

Reply:
> Should I do both?
>
> 1. User option for enabling or disabling escape codes
> 2. Check `stdout` with `isatty`

Yes. The default will be #2 ("auto"), and we allow the user to overwrite the decision with #1.

> Also should we write different tests for escaped characters?

In general, we should test what we want to work.

A "proper" test involves writing a terminal emulator. Not worth it. A "hacky" test involves string comparisons, in addition to manual testing, which is fine IMO.